corehq/apps/export/forms.py

```diff
@@ -1,139 +1,336 @@
-            (module.case_type, module.case_type)
-            for app in apps if hasattr(app, 'modules')
-            for module in app.modules
-            if module.case_type
-        ]
+    def __init__(self, *args, **kwargs):
+        super(CreateCaseExportTagForm, self).__init__(*args, **kwargs)
 
         self.helper = FormHelper()
-        self.helper.form_id = "create-export-form"
-        self.helper.form_class = "form-horizontal"
+        self.helper.form_tag = False
+        self.helper.label_class = 'col-sm-2'
+        self.helper.field_class = 'col-sm-10'
 
         self.helper.layout = crispy.Layout(
-            crispy.Fieldset(
-                _('Select Case Type'),
+            crispy.Field(
+                'application',
+                placeholder=_("Select Application"),
+                ng_model="createForm.application",
+                ng_change="updateCaseTypes()",
+                ng_required="true",
+            ),
+            crispy.Field(
+                'case_type',
+                placeholder=_("Select Case Type"),
+                ng_model="createForm.case_type",
+                ng_disabled="!createForm.application",
+                ng_required="true",
+            ),
+        )
+
+
+class FilterExportDownloadForm(forms.Form):
+    _export_type = 'all'  # should be form or case
+
+    _USER_MOBILE = 'mobile'
+    _USER_DEMO = 'demo_user'
+    _USER_ADMIN = 'admin'
+    _USER_UNKNOWN = 'unknown'
+    _USER_SUPPLY = 'supply'
+
+    _USER_TYPES_CHOICES = [
+        (_USER_MOBILE, ugettext_noop("All Mobile Workers")),
+        (_USER_DEMO, ugettext_noop("Demo User")),
+        (_USER_ADMIN, ugettext_noop("Admin User")),
+        (_USER_UNKNOWN, ugettext_noop("Unknown Users")),
+        (_USER_SUPPLY, ugettext_noop("CommCare Supply")),
+    ]
+    type_or_group = forms.ChoiceField(
+        label=ugettext_noop("User Types or Group"),
+        required=False,
+        choices=(
+            ('type', ugettext_noop("User Types")),
+            ('group', ugettext_noop("Group")),
+        )
+    )
+    user_types = forms.MultipleChoiceField(
+        label=ugettext_noop("Select User Types"),
+        widget=Select2MultipleChoiceWidget,
+        choices=_USER_TYPES_CHOICES,
+        required=False,
+    )
+    group = forms.CharField(
+        label=ugettext_noop("Select Group"),
+        required=False,
+    )
+    date_range = forms.CharField(
+        label=ugettext_noop("Date Range"),
+        required=True,
+        widget=DateRangePickerWidget(),
+    )
+
+    def __init__(self, domain, timezone, *args, **kwargs):
+        self.domain = domain
+        self.timezone = timezone
+        if not isinstance(self.domain, Domain):
+            self.domain = Domain.get_by_name(self.domain)
+
+        super(FilterExportDownloadForm, self).__init__(*args, **kwargs)
+
+        if not self.domain.uses_locations:
+            self.fields['user_types'].choices = self._USER_TYPES_CHOICES[:-1]
```

Comment:
(small) a comment explaining this, or even better, changing it to a function with a name that explains what it does, might be nice